NightConfig Flaky crash

Only observed on Neoforge 1.20.4 up to now.
https://forums.minecraftforge.net/topic/141171-1201-crash-on-game-load-config-issue/
```
"This is a known rare issue with the config lib Forge uses, caused by concurrent modification. The author of NightConfig has been made aware and we're waiting on a fix from them.

The bug triggers more often when config files are left open in editors while the game is launching or running, or multiple instances of the game are started, however it can unfortunately also happen at random without doing either of those things."
```
Odd as we do not have multiple instances open on the same machine when running the matrix workflow with all the instances?

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "net.neoforged.fml.loading.FMLConfig$ConfigValue.getConfigValue(com.electronwill.nightconfig.core.file.CommentedFileConfig)" is null
	at net.neoforged.fml.earlydisplay.DisplayWindow.setupMinecraftWindow(DisplayWindow.java:523) ~[earlydisplay-2.0.17.jar%23126!/:2.0] {}
	at net.neoforged.fml.loading.ImmediateWindowHandler.setupMinecraftWindow(ImmediateWindowHandler.java:68) ~[loader-2.0.17.jar%23127!/:2.0] {}
	at com.mojang.blaze3d.platform.Window.<init>(Window.java:87) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.VirtualScreen.newWindow(VirtualScreen.java:21) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:classloading}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:475) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:206) ~[neoforge-20.4.237.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "net.neoforged.fml.loading.FMLConfig$ConfigValue.getConfigValue(com.electronwill.nightconfig.core.file.CommentedFileConfig)" is null
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?] {}
	at java.util.concurrent.FutureTask.get(FutureTask.java:205) ~[?:?] {}
	at net.neoforged.fml.earlydisplay.DisplayWindow.setupMinecraftWindow(DisplayWindow.java:521) ~[earlydisplay-2.0.17.jar%23126!/:2.0] {}
	... 21 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "net.neoforged.fml.loading.FMLConfig$ConfigValue.getConfigValue(com.electronwill.nightconfig.core.file.CommentedFileConfig)" is null
	at net.neoforged.fml.loading.FMLConfig.getBoolConfigValue(FMLConfig.java:137) ~[loader-2.0.17.jar%23127!/:2.0] {}
	at net.neoforged.fml.earlydisplay.DisplayWindow.initRender(DisplayWindow.java:235) ~[earlydisplay-2.0.17.jar%23126!/:2.0] {}
	at net.neoforged.fml.earlydisplay.DisplayWindow.lambda$start$9(DisplayWindow.java:287) ~[earlydisplay-2.0.17.jar%23126!/:2.0] {}
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?] {}
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?] {}
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?] {}
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?] {}
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?] {}
	at java.lang.Thread.run(Thread.java:840) ~[?:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.neoforged.fml.earlydisplay.DisplayWindow.setupMinecraftWindow(DisplayWindow.java:523) ~[earlydisplay-2.0.17.jar%23126!/:2.0] {}
	at net.neoforged.fml.loading.ImmediateWindowHandler.setupMinecraftWindow(ImmediateWindowHandler.java:68) ~[loader-2.0.17.jar%23127!/:2.0] {}
	at com.mojang.blaze3d.platform.Window.<init>(Window.java:87) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.renderer.VirtualScreen.newWindow(VirtualScreen.java:21) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:classloading}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:475) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
-- Initialization --
Details:
	Modules: 
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:206) ~[neoforge-20.4.237.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0] {}
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
```

--- a/.github/actions/local-action/action.yml
+++ b/.github/actions/local-action/action.yml
@@ -57,7 +57,7 @@ inputs:
   hmc-version:
     description: 'The version of headlessmc to download.'
     required: false
-    default: '1.10.2'
+    default: '2.1.0'
 
 runs:
   using: "composite"
@@ -109,7 +109,7 @@ runs:
       shell: bash
     - name: Download mc-runtime-test jar ${{ inputs.mc-runtime-test }}
       if: ${{ inputs.mc-runtime-test != 'none' }}
-      run: wget -O run/mods/mc-runtime-test-${{ inputs.mc }}-2.3.1-${{ inputs.mc-runtime-test }}-release.jar https://github.com/3arthqu4ke/mc-runtime-test/releases/download/2.3.1/mc-runtime-test-${{ inputs.mc }}-2.3.1-${{ inputs.mc-runtime-test }}-release.jar
+      run: wget -O run/mods/mc-runtime-test-${{ inputs.mc }}-2.4.0-${{ inputs.mc-runtime-test }}-release.jar https://github.com/3arthqu4ke/mc-runtime-test/releases/download/2.4.0/mc-runtime-test-${{ inputs.mc }}-2.4.0-${{ inputs.mc-runtime-test }}-release.jar
       shell: bash
     - name: Download fabric-api jar ${{ inputs.fabric-api }}
       if: ${{ inputs.fabric-api != 'none' }}

--- a/.github/workflows/fabric.yml
+++ b/.github/workflows/fabric.yml
@@ -30,7 +30,7 @@ jobs:
             -   name: Configure Dummy Assets
                 run: echo hmc.assets.dummy=true >> HeadlessMC/config.properties
             -   name: Get HeadlessMC
-                run: wget -O headlessmc-launcher.jar https://github.com/3arthqu4ke/headlessmc/releases/download/1.10.2/headlessmc-launcher-1.10.2.jar
+                run: wget -O headlessmc-launcher.jar https://github.com/3arthqu4ke/headlessmc/releases/download/2.1.0/headlessmc-launcher-2.1.0.jar
             -   name: Download 1.20.4
                 run: java -jar headlessmc-launcher.jar --command download 1.20.4
             -   name: Download Fabric 1.20.4

--- a/.github/workflows/run-gametests-latest.yml
+++ b/.github/workflows/run-gametests-latest.yml
@@ -58,12 +58,12 @@ jobs:
         run: mkdir -p run/mods
       - name: Copy mod files
         run: |
-          cp gametest-jars/clientgametest-1.20.6-2.3.1-fabric.jar run/mods
-          cp gametest-jars/clientgametest-1.20.6-2.3.1-neoforge.jar run/mods
-          cp gametest-jars/clientgametest-1.20.6-2.3.1-lexforge.jar run/mods
-          cp latest-jars/mc-runtime-test-1.20.6-2.3.1-fabric-release.jar run/mods
-          cp latest-jars/mc-runtime-test-1.20.6-2.3.1-lexforge-release.jar run/mods
-          cp latest-jars/mc-runtime-test-1.20.6-2.3.1-neoforge-release.jar run/mods
+          cp gametest-jars/clientgametest-1.20.6-2.4.0-fabric.jar run/mods
+          cp gametest-jars/clientgametest-1.20.6-2.4.0-neoforge.jar run/mods
+          cp gametest-jars/clientgametest-1.20.6-2.4.0-lexforge.jar run/mods
+          cp latest-jars/mc-runtime-test-1.20.6-2.4.0-fabric-release.jar run/mods
+          cp latest-jars/mc-runtime-test-1.20.6-2.4.0-lexforge-release.jar run/mods
+          cp latest-jars/mc-runtime-test-1.20.6-2.4.0-neoforge-release.jar run/mods
       - name: Run game for version
         id: local-action
         uses: ./.github/actions/local-action

--- a/.github/workflows/run-specific-reusable.yml
+++ b/.github/workflows/run-specific-reusable.yml
@@ -55,7 +55,7 @@ on:
       hmc-version:
         description: 'The version of headlessmc to download.'
         required: false
-        default: '1.10.2'
+        default: '2.1.0'
         type: string
 
 jobs:

--- a/.github/workflows/run-specific-xvfb.yml
+++ b/.github/workflows/run-specific-xvfb.yml
@@ -43,7 +43,7 @@ on:
       hmc-version:
         description: 'The version of headlessmc to download.'
         required: false
-        default: '1.10.2'
+        default: '2.1.0'
         type: string
 
 jobs:

--- a/.github/workflows/run-specific.yml
+++ b/.github/workflows/run-specific.yml
@@ -43,7 +43,7 @@ on:
       hmc-version:
         description: 'The version of headlessmc to download.'
         required: false
-        default: '1.10.2'
+        default: '2.1.0'
         type: string
 
 jobs:

--- a/README.md
+++ b/README.md
@@ -64,7 +64,7 @@ jobs:
         run: mkdir -p run/mods && cp build/libs/<your-mod>.jar run/mods
       # Call this Action to run the client
       - name: Run the MC client
-        uses: 3arthqu4ke/mc-runtime-test@2.3.1
+        uses: 3arthqu4ke/mc-runtime-test@2.4.0
         with:
           mc: 1.20.4
           modloader: fabric

--- a/README.md
+++ b/README.md
@@ -77,7 +77,12 @@ jobs:
           mc-runtime-test: fabric
           java: 17
 ```
-An example workflow in action can be found [here](https://github.com/3arthqu4ke/hmc-optimizations/blob/1.20.4/.github/workflows/run-fabric.yml).
+An example workflow in action can be found
+[here](https://github.com/3arthqu4ke/hmc-optimizations/blob/1.20.4/.github/workflows/run-fabric.yml).
+An example for a large matrix workflow
+which tests 20 different versions of Minecraft
+at once can be found 
+[here](https://github.com/3arthqu4ke/hmc-specifics/blob/main/.github/workflows/run-matrix.yml).
 
 # Inputs
 - `mc`: The MC version to use, e.g. `1.20.4`.

--- a/README.md
+++ b/README.md
@@ -33,9 +33,12 @@ registered.
 
 Mc-Runtime-Test currently supports the following Minecraft versions and modloaders:
 You can configure it to use any other version, but in that case you need to set `mc-runtime-test` to `none` and provide another way for the game to exit, or the workflow will run indefinitely.
+
+<div align="center">
+  
 | Version  | Forge | Fabric | NeoForge | 
 | :-: | :-: | :-: | :-: |
-| 1.21  | :white_check_mark:  | :white_check_mark:  | :white_check_mark: |
+| 1.21 - 1.21.1  | :white_check_mark:  | :white_check_mark:  | :white_check_mark: |
 | 1.20.2 - 1.20.6  | :white_check_mark:  | :white_check_mark:  | :white_check_mark: |
 | 1.20.1  | :white_check_mark:  | :white_check_mark:  | :warning:  |
 | 1.19 - 1.19.4  | :white_check_mark:  | :white_check_mark:  | - |
@@ -45,6 +48,8 @@ You can configure it to use any other version, but in that case you need to set 
 | 1.12.2  | :white_check_mark:  | :warning:  | - |
 | 1.8.9  | :white_check_mark:  | :warning:  | - |
 | 1.7.10  | :white_check_mark:  | :warning:  | - |
+
+</div>
 
 Versions marked with :warning: have not been tested yet, due to not being supported by HeadlessMC, e.g. fabric legacy versions.
 

--- a/action.yml
+++ b/action.yml
@@ -57,7 +57,7 @@ inputs:
   hmc-version:
     description: 'The version of headlessmc to download.'
     required: false
-    default: '1.10.2'
+    default: '2.1.0'
 
 runs:
   using: "composite"
@@ -106,7 +106,7 @@ runs:
       shell: bash
     - name: Download mc-runtime-test jar ${{ inputs.mc-runtime-test }}
       if: ${{ inputs.mc-runtime-test != 'none' }}
-      run: wget -O run/mods/mc-runtime-test-${{ inputs.mc }}-2.3.1-${{ inputs.mc-runtime-test }}-release.jar https://github.com/3arthqu4ke/mc-runtime-test/releases/download/2.3.1/mc-runtime-test-${{ inputs.mc }}-2.3.1-${{ inputs.mc-runtime-test }}-release.jar
+      run: wget -O run/mods/mc-runtime-test-${{ inputs.mc }}-2.4.0-${{ inputs.mc-runtime-test }}-release.jar https://github.com/3arthqu4ke/mc-runtime-test/releases/download/2.4.0/mc-runtime-test-${{ inputs.mc }}-2.4.0-${{ inputs.mc-runtime-test }}-release.jar
       shell: bash
     - name: Download fabric-api jar ${{ inputs.fabric-api }}
       if: ${{ inputs.fabric-api != 'none' }}

--- a/api/gradle.properties
+++ b/api/gradle.properties
@@ -1,1 +1,1 @@
-project_version = 2.3.1
+project_version = 2.4.0

